src/notebooks/03_plot_tsne_videos.ipynb

```diff
@@ -0,0 +1,321 @@
+{
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "id": "4f5c8046-b184-4583-88c8-e4c4d74a2272",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import sys\n",
+    "import os"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "id": "3f3794e3-dd59-43ff-80c7-8f1d324df574",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sys.path.append(os.path.abspath('../../app'))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "id": "decb3603-c5b2-4b9b-8f7e-2e5629a03b12",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "/usr/app/venv/lib/python3.11/site-packages/tqdm/auto.py:21: TqdmWarning: IProgress not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html\n",
+      "  from .autonotebook import tqdm as notebook_tqdm\n"
+     ]
+    }
+   ],
+   "source": [
+    "from huggingface_hub import hf_hub_download\n",
+    "from core.feluda import Feluda\n",
+    "from core.models.media_factory import VideoFactory\n",
+    "import cv2\n",
+    "import tarfile\n",
+    "import matplotlib.pyplot as plt\n",
+    "from matplotlib.offsetbox import OffsetImage, AnnotationBbox"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "a5bcbdae-8910-47c3-af07-67131cf76ab2",
+   "metadata": {},
+   "source": [
+    "\n",
+    "### Dataset Structure Breakdown\n",
+    "\n",
+    "- **UCF101_subset/**: The root directory containing the dataset.\n",
+    "  - **train/**: Contains training samples.\n",
+    "    - Each subdirectory under `train/` corresponds to a specific action class:\n",
+    "      - `BenchPress`\n",
+    "      - `BasketballDunk`\n",
+    "      - `BalanceBeam`\n",
+    "      - `ApplyLipstick`\n",
+    "      - `BabyCrawling`\n",
+    "      - `ApplyEyeMakeup`\n",
+    "      - `Archery`\n",
+    "      - `BandMarching`\n",
+    "      - `BaseballPitch`\n",
+    "      - `Basketball`\n",
+    "  - **test/**: Same as train.\n",
+    "  - **val/**: Same as train.\n",
+    "\n",
+    "We'll take train subset as our example here."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "id": "aa8c9e2d-066e-43b7-add8-2b5d5293d1c9",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Downloading and extracting \n",
+    "\n",
+    "dataset_name = \"UCF101_subset/train\"\n",
+    "hf_dataset_identifier = \"sayakpaul/ucf101-subset\"\n",
+    "filename = \"UCF101_subset.tar.gz\"\n",
```

Comment:
does the `tar.gz` file gets deleted in the end?

Reply:
This tar.gz files get stores in python cache.
'/home/python/.cache/huggingface/hub/datasets--sayakpaul--ucf101-subset/snapshots/b9984b8d2a95e4a1879e1b071e9433858d0bc24a/UCF101_subset.tar.gz'.
I don't think we need to specifically remove these, but I may be wrong. What do you think?